resource/resource_graph_export.go

```diff
@@ -11,6 +12,61 @@ import (
 	"go.viam.com/rdk/utils"
 )
 
+const snapshotLimit = 500
+
+// Visualizer stores a history resource graph DOT snapshots.
+type Visualizer struct {
+	snapshots list.List
+}
+
+// SnapshotInfo contains a DOT snapshot string along with metadata.
+type SnapshotInfo struct {
```

Comment:
Oh wait, I forgot a real feature request! Can we add the timestamp a snapshot was recorded here and render it? If we're getting into scope creep for this ticket don't worry about it. Happy to wait for follow-up PR or just pitch in myself after this.